### ISSUE TITLE
Seeding and multi-GPU training

### ISSUE THREAD

**Issue by tomasgeffner** [Lightning-AI/pytorch-lightning#20188]
### 📚 Documentation

I'm training a model in a multi GPU environment using the DDP strategy. Looking [here](https://pytorch-lightning.readthedocs.io/en/0.9.0/multi_gpu.html#data-parallel) I see that it is important to call `L.seed_everything(...)` to make sure the model is initialized the same way across devices. However [here](https://github.com/Lightning-AI/pytorch-lightning/issues/12986#issuecomment-1118737218) it says that this is not needed. I tried a test run on my environment and noted that even without calling `seed_everything` I get that the model is initialized with the same weights across devices, which makes me think it is the latter. Is this correct?

And quick follow-up. If I wanted to set a different seed for each device, how would I go about it? Just normal `seed_everything` but with a different seed value for each process (e.g. using `self.global_rank` inside the module)?

Thanks

cc @lantiga @borda

**Comment by jlotthammer** [comment]
Also very interested in an answer to this as I'm seeing conflicting documentation online here too - e.g., on [WANDB](https://docs.wandb.ai/guides/integrations/lightning):

```
def main():
    # Setting all the random seeds to the same value.
    # This is important in a distributed training setting.
    # Each rank will get its own set of initial weights.
    # If they don't match up, the gradients will not match either,
    # leading to training that may not converge.
    pl.seed_everything(1)
```

**Comment by SkafteNicki** [comment]
Hi @tomasgeffner and @jlotthammer, thanks for raising this issue and commenting on it.

The TLDR is that your right. Models do get initialized with the same weights across devices in DDP even without calling
`seed_everything`. Without calling `seed_everything` lightning still:

1. Make sure parameters are in `sync`
2. Make sure that the same model is initialized across all devices without any manual seeding

However, we still recommend using `seed_everything` because it adds:

1. Data loading reproducibility
2. Consistency in layer behavior e.g. dropout and batch norm randomness
3. Just overall reproducibility of experiments

Regarding the link to wandb: we are not in charge of their documentation. I can see the confusion but please refer to our own documentation for answers.
